lib/Sema/CSSimplify.cpp

```diff
@@ -3825,8 +3822,41 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
 
     // FIXME(diagnostics): If there were no viable results, but there are
     // unviable ones, we'd have to introduce fix for each specific problem.
-    if (!result.UnviableCandidates.empty())
+    if (!result.UnviableCandidates.empty()) {
+      // Check if we have unviable candidates whose reason for rejection
+      // is either UR_InstanceMemberOnType or UR_TypeMemberOnInstance
+      
+      // If we do, then allow them
+      if (llvm::all_of(result.UnviableCandidates,
+                       [&](const std::pair<OverloadChoice,
+                           MemberLookupResult::UnviableReason> &e) {
+        return e.second == MemberLookupResult::UR_InstanceMemberOnType ||
+               e.second == MemberLookupResult::UR_TypeMemberOnInstance;
+      })) {
+        
+        auto *fix = AllowTypeOrInstanceMember::create(*this, baseTy, member,
+                                                      getConstraintLocator(locator));
+        
+        if (recordFix(fix)) {
+          // We couldn't record a fix, so bail out
+          return SolutionKind::Error;
+        }
+        
+        // The fix was successful, so let's add it to the overload set
+        // and return the solution as solved
+        auto overloadChoices = SmallVector<OverloadChoice, 4>();
```

Comment:
Nit: I haven't actually seen this style anywhere in our repository, might be worth to change it to `SmallVector<OverloadChoice, 4> choices;` instead.